mm2src/coins/utxo/utxo_builder/utxo_coin_builder.rs

```diff
@@ -311,23 +358,22 @@ pub trait UtxoFieldsWithHardwareWalletBuilder: UtxoCoinBuilderCommonOps {
         let recently_spent_outpoints = AsyncMutex::new(RecentlySpentOutPoints::new(my_script_pubkey));
```

Comment:
I just have taken a look how RecentlySpentOutPoints works.
One of its tasks is that it allows to immediately spend an output sent to self ('my_address') from a just broadcasted transaction. This is needed because the electrum provider may not know immediately about just sent txns. Basically this was added for 'change' outputs. But I think the current RecentlySpentOutPoints implementation (when it is created only with the initially activated 'my_address') does not cover cases with HD accounts: 
- when a user withdraws from some HD account the change is sent to this 'from' account (not to 'my_address'). 
- another case is when a user sends coins from one HD account to another his account.

(Not sure if we need this fixed though. Maybe it's okay for the user to just wait for a couple of min until the provider db is actualised)

Reply:
It will be easier done in this PR https://github.com/KomodoPlatform/komodo-defi-framework/pull/2053 or after it's merged where `UnspentInfo` has the `script_pubkey`. Add this to the checklist here https://github.com/KomodoPlatform/komodo-defi-framework/issues/1838#issuecomment-1675467570